test/Spec/BotPlutusInterface/Balance.hs

```diff
@@ -105,3 +165,131 @@ addUtxosForChange = do
   case ebalancedTx of
     Left e -> assertFailure (Text.unpack e)
     Right balancedTx -> txInputs <$> balancedTx @?= Right (Set.fromList [txIn1, txIn2])
+
+dontAddChangeToDatum :: Assertion
+dontAddChangeToDatum = do
+  let scrTxOut' =
+        ScriptChainIndexTxOut
+          valAddr
+          (Right validator) -- (valHash, Just validator)
+          (Right scrDatum) -- (scrDatumHash, Just scrDatum)
+          scrValue
+      scrTxOut = Ledger.toTxOut scrTxOut'
+      usrTxOut' =
+        PublicKeyChainIndexTxOut
+          pkhAddr3
+          (Ada.lovelaceValueOf 1_001_000)
+      usrTxOut = Ledger.toTxOut usrTxOut'
+      -- initState :: MockContractState ()
+      initState =
+        def & utxos .~ [(txOutRef6, scrTxOut), (txOutRef7, usrTxOut)]
+          & contractEnv .~ contractEnv'
+      pabConf :: PABConfig
+      pabConf = def {pcOwnPubKeyHash = pkh3}
+      -- contractEnv' :: ContractEnvironment ()
+      contractEnv' = def {cePABConfig = pabConf}
+
+      -- Input UTxOs:
+      -- UTxO 1:
+      -- - From: User
+      -- - Amt : 1.001 ADA
+      -- UTxO 2:
+      -- - From: Script
+      -- - Amt : 0.5 ADA + 200 Tokens
+      --
+      -- Output UTxOs:
+      -- UTxO 1:
+      -- - To : User
+      -- - Amt: 1 ADA
+      -- UTxO 2:
+      -- - To : Script
+      -- - Amt: 1.0005 Ada + 200 Token
```

Comment:
It looks like it should be `500500` Ada, as script had 500_000 Ada and then 500 more were paid.

Reply:
Yeah, you're right. Fixed.